src/PublicSuffix.php

```diff
@@ -283,9 +310,34 @@ public function toUnicode()
         }
 
         $clone = clone $this;
-        $clone->publicSuffix = $this->idnToUnicode($this->publicSuffix);
+        $clone->publicSuffix = $this->idnToUnicode($this->publicSuffix, $this->unicodeIDNAOption);
         $clone->labels = array_reverse(explode('.', $clone->publicSuffix));
 
         return $clone;
     }
+    
+    /**
+     * @return array
+     */
+    public function getIDNAOptions(): array
+    {
+        return [$this->asciiIDNAOption, $this->unicodeIDNAOption];
+    }
+    
+    /**
+     * return true if domain contains deviation characters.
+     * @see http://unicode.org/reports/tr46/#Transition_Considerations
+     * @return bool
+     **/
+    public function isTransitionalDifferent(): bool
```

Comment:
no `try.. catch` should be needed on a getter method. This depends on how `$isTransitionalDifferent` setter is defined.

Reply:
Finally, your point about setter definition is not clear for me

Reply:
`$isTransitionalDifferent` is defined in the `Domain` and `PublicSuffix` VO but is used in the trait, which is bad. 
So you either:

- defined it in the trait directly and any classes which uses this trait will automatically have at least the property.
- or you create a setter method in the trait and you use this method to determine the `$isTransitionalDifferent` in each individual VO but without the property appearing directly in the trait.

Hope this clarify my comment.